kernel/kernel-api/src/main/java/io/delta/kernel/internal/skipping/DataSkippingUtils.java

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.skipping;
+
+import java.util.*;
+
+import io.delta.kernel.client.TableClient;
+import io.delta.kernel.data.ColumnVector;
+import io.delta.kernel.data.ColumnarBatch;
+import io.delta.kernel.data.FilteredColumnarBatch;
+import io.delta.kernel.expressions.*;
+import io.delta.kernel.internal.actions.AddFile;
+import io.delta.kernel.types.*;
+import static io.delta.kernel.internal.InternalScanFileUtils.SCAN_FILE_SCHEMA;
+import static io.delta.kernel.internal.util.Preconditions.checkArgument;
+import static java.lang.String.format;
+
+public class DataSkippingUtils {
+
+    private static final int ADD_FILE_ORDINAL = SCAN_FILE_SCHEMA.indexOf("add");
+    private static final int ADD_FILE_STATS_ORDINAL = AddFile.SCHEMA_WITH_STATS.indexOf("stats");
+
+    /**
+     * Given the table schema return the expected statistics schema.
+     */
+    public static StructType getStatsSchema(StructType dataSchema) {
+        return StatsSchemaHelper.getStatsSchema(dataSchema);
+    }
+
+    /**
+     * Given a {@code FilteredColumnarBatch} of scan files and the statistics schema to parse,
+     * return the parsed JSON stats from the scan files.
+     */
+    public static ColumnarBatch parseJsonStats(
+        TableClient tableClient, FilteredColumnarBatch scanFileBatch, StructType statsSchema) {
+        ColumnVector statsVector = scanFileBatch.getData()
+            .getColumnVector(ADD_FILE_ORDINAL)
+            .getChild(ADD_FILE_STATS_ORDINAL);
+        return tableClient.getJsonHandler()
+            .parseJson(statsVector, statsSchema, scanFileBatch.getSelectionVector());
```

Comment:
Not for this PR, but I feel like there is an opportunity for future optimizations here. just like we expressions evaluation has two steps - create the evaluator (with 1 time setup stuff), and then execute the evaluator (efficiently again and again)... the json parser could probably use a similar setup / execute break down.